### PR TITLE
 Add docs for Many to Many relation

### LINKED ISSUE #73: Many to many Relationships
Thank you for your awesome work on this library!

My question:
Is there support (planed?) for many-to-many relationships?

Example:
I have a cookbook table. A cookbook belongs to a specific user.
I have a recipe table.
A recipe can belong to many cookbooks (a user can „favorite“ the recipe in his personal cookbook).

This could be solved using a associative/pivot table containing foreign keys to the cookbook table and the recipe table.

### PR DESCRIPTION
**Pull request by Kabangi**
Closes #73

### CODE REVIEW

File: docs/Model.md [hunk 1]
```diff
@@ -111,7 +111,7 @@ class Comment extends Model {
 
 **➡️ Learn more:** [Relation API](./Relation.md)
 
-### Children
+### To-Many relation
```
Comment (radex) [inline docs/Model.md]:
```suggestion
### Children (To-Many relation)
```

File: docs/Model.md [hunk 2]
```diff
@@ -132,6 +132,66 @@ Pass the _table name_ of the related records as an argument to `@children`. The
 
 **Note:** You must define a `has_many` association in `static associations` for this to work
 
+### Many-To-Many Relation
```
Comment (radex) [inline docs/Model.md]: I suggest moving it down to the "Advanced" section

File: docs/Model.md [hunk 3]
```diff
@@ -132,6 +132,66 @@ Pass the _table name_ of the related records as an argument to `@children`. The
...
+
+If for instance, our app `Post`s can be authored by many `User`s and a user can author many `Post`s. We would create such a relation following these steps:- 
+
+1. Create a pivot schema and model that both the `User` model and `Post` model has association to; say `PostUser`
```
Comment (radex) [inline docs/Model.md]: PostUser → PostAuthor perhaps? i think that would be more clear

File: docs/Model.md [hunk 4]
```diff
@@ -132,6 +132,66 @@ Pass the _table name_ of the related records as an argument to `@children`. The
...
+import {lazy } from '@nozbe/watermelondb/decorators'
+
+class Post extends Model {
+  static table = 'posts'
+  static associations = {
+    post_users: { type: 'has_many', foreignKey: 'post_id' },
+  }
+
+  @lazy
+  posts = this.collections
```
Comment (radex) [inline docs/Model.md]:
```suggestion
  authors = this.collections
```

File: docs/Model.md (hunk 1 above)
Comment (radex) [reply docs/Model.md]: I think that's a little more clear

**Comment by radex** [comment]
> Also, I think we should migrate the relation docs from model docs to Relation docs for better visibility. If this okay I can go ahead and do that.

Ah, you're right. I think the very basics ( @children, relation) should stay as is, but the advanced details (like many-to-many ) should be moved to Relation.md

File: docs/Model.md (hunk 1 above)
Comment (Kabangi) [reply docs/Model.md]: Yes much better

File: docs/Model.md (hunk 2 above)
Comment (Kabangi) [reply docs/Model.md]: > > Also, I think we should migrate the relation docs from model docs to Relation docs for better visibility. If this okay I can go ahead and do that.
> 
> Ah, you're right. I think the very basics ( @children, relation) should stay as is, but the advanced details (like many-to-many ) should be moved to Relation.md

Yes, I think this is much better. I will move the docs there then.

**Comment by Kabangi** [comment]
@radex Updated

**Comment by radex** [comment]
PS. Thanks again @Kabangi for contributing. Would you like to play a little more with Watermelon's code or docs? I have some a list of easy things to tackle: https://github.com/Nozbe/WatermelonDB/issues/141 if you'd like to contribute. (And I can offer to send you some Watermelon stickers ;) )

File: docs/Model.md [hunk 5]
```diff
@@ -132,7 +132,7 @@ Pass the _table name_ of the related records as an argument to `@children`. The
 
 **Note:** You must define a `has_many` association in `static associations` for this to work
 
-**➡️ Learn more:** [Queries](./Query.md)
+**➡️ Learn more:** [Relation API](./Relation.md)
```
[the remark this answers is not in the document]
Comment (Kabangi) [reply docs/Model.md]: I will revert, I missed the context here a bit.

**Comment by Kabangi** [comment]
@radex Am happy to do both code and docs. I will be looking to send more PR's next week

**Pull request merged**